### ISSUE TITLE
Highlight objects that should have an index

### ISSUE THREAD

**Issue by ggnall** [MaterializeInc/materialize#19935]
## Problem 
It's not obvious to users when the objects they are querying from are missing an index, especially when they are querying built-in views.

## Solution
- A view/ui for surfacing objects that are commonly fully scanned from Persist. 

## Notes

[Slack thread](https://materializeinc.slack.com/archives/C056M3001Q9/p1686577269097699)

**Comment by ggevay** [comment]
Currently, it's hard to even know how exactly the used indexes are being used (scan or lookup) and in which part of the query: https://github.com/MaterializeInc/materialize/issues/15242. We should probably fix that before this (potentially, by https://github.com/MaterializeInc/materialize/issues/16598).

**Comment by ggevay** [comment]
Created an issue for the situation when there is already a wider index than a literal constraint: https://github.com/MaterializeInc/materialize/issues/20768

**Comment by ggevay** [comment]
Suggest an index to the user when two dataflows are arranging the same thing: https://github.com/MaterializeInc/materialize/issues/20802

**Comment by ggevay** [comment]
This issue was a bit wide, and many other issues under the Guardrails epic (https://github.com/MaterializeInc/console/issues/549) overlapped with it. I refocused this one on specifically those situations where we need to gather some statistics on ad hoc queries to realize that something would be indexed. I went ahead and edited the title, if that's ok @ggnall.

**Comment by ggnall** [comment]
Yes, thank you for driving this!On Aug 31, 2023, at 3:20 PM, Gábor E. Gévay ***@***.***> wrote:﻿
This issue was a bit wide, and many other issues under the Guardrails epic (MaterializeInc/console#549) overlapped with it. So, I refocused this one on specifically those situations where we need to gather some statistics on ad hoc queries to realize that something would be indexed. I went ahead and edited the title, if that's ok @ggnall.
